internal/pkg/agent/application/managed_mode.go

```diff
@@ -135,14 +135,30 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 		close(retrierRun)
 	}()
 
+	// Gather errors from the dispatcher and pass to the error channel.
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case err := <-actionDispatcher.Errors():
+				m.errCh <- err // err is one or more failures from dispatching an action
+			}
+		}
+	}()
+
 	actions := m.stateStore.Actions()
 	stateRestored := false
 	if len(actions) > 0 && !m.wasUnenrolled() {
 		// TODO(ph) We will need an improvement on fleet, if there is an error while dispatching a
 		// persisted action on disk we should be able to ask Fleet to get the latest configuration.
 		// But at the moment this is not possible because the policy change was acked.
-		if err := store.ReplayActions(ctx, m.log, actionDispatcher, actionAcker, actions...); err != nil {
+		m.log.Info("restoring current policy from disk")
+		actionDispatcher.Dispatch(ctx, actionAcker, actions...)
+		select {
+		case err := <-actionDispatcher.Errors():
```

Comment:
Why is this still here? Isn't the new goroutine started above going to read and handle these errors?

Reply:
changed